src/main/java/org/logstash/beats/BeatsHandler.java

```diff
@@ -56,28 +62,29 @@ public void channelRead0(ChannelHandlerContext ctx, Batch batch) throws Exceptio
             }
         }
         ctx.flush();
-        processing.compareAndSet(true, false);
-
+        ctx.channel().attr(PROCESSING_BATCH).set(false);
     }
 
+    /*
+     * Do not propagate the ssl handshake exception down to the ruby layer
+     * handle it locally and close the connection if the channel is still active.
+     * Calling `onException` will flush the content of the codec's buffer, this call may block the thread
+     * in the event loop until completion, this should only affect LS 5 because it still supporting
+     * the multiline codec, v6 drop support for buffering codec in the input.
+     *
+     * For v5, I cannot simply drop the content of the buffer because this will create data loss, because multiline
+     * content can overlap Filebeat transmission windows, we were recommending multiline at the source in v5 and in v6
+     * we enforce it.
+     */
     @Override
     public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) {
-        logger.info(format("Exception: " + cause.getMessage()));
-        messageListener.onException(ctx, cause);
         ctx.close();
```

Comment:
Same change as in https://github.com/logstash-plugins/logstash-input-beats/issues/259#issuecomment-349314102